doc/source/dev/missing-bits.rst

```diff
@@ -0,0 +1,260 @@
+
+.. [*] This function accepts positional arguments only, but the docstring
+       documents the argument as `lambda`.
+
+
+Implicit string continuation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+When an implicit continuation is used to create a long literal string, the
+string is generally split at a space, and the space is included in the
+second part of the string.
+
+  Yes::
+
+      long_string = ("This is a very long string that demonstrates"
+                     " this guideline.")
+
+  No::
+
+      long_string = ("This is a very long string that demonstrates "
+                     "this guideline.")
+
+
+Documentation Guidelines
+------------------------
+
+Use of LaTeX
+~~~~~~~~~~~~
+[TBD.  In the past, we had an unwritten rule that LaTex markup should be
+restricted to the Notes section.  We don't follow that rule these days.
+Should we?]
+
+
+Use "must", not "should"
+~~~~~~~~~~~~~~~~~~~~~~~~
+When specifying a required condition on the input parameters, the
+word "must" is preferable to "should".  For many English speakers,
+"must" implies a stronger constraint than "should",  e.g. "I must
+have oxygen to live" versus "I should exercise more".
+
+    Yes::
+
+            Parameters
+            ----------
+            x : float
+                x must be nonnegative.
+
+    No::
+
+            Parameters
+            ----------
+            x : float
+                x should be nonnegative.
+
+
+Use of the 'versionadded' markup
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+* For a new function, the 'versionadded' markup goes in the "Notes" section,
+  *not* in the description at the beginning of the docstring.
+* For a new argument added to an existing function,  two locations have been
+  used for the the 'versionadded' markup, [TBD: which is preferred?]:
+
+  * At the end of the description of the argument in the "Parameters" section
+  * In the "Notes" section.  In this case, the `versionadded` markup
+    wouldn't be used.  Instead, the new addition is noted with a plain
+    text comment.
+
+
+Citing wikipedia articles in the "References" section
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+It is acceptable to use wikipedia articles as references.
+When creating the citation for the reference, include the article title,
+the name "Wikipedia" (similar to how one gives a journal title), and the
+URL.
+
+    Yes::
+
+        .. [1] "Zeta Distribution", Wikipedia,
+               https://en.wikipedia.org/wiki/Zeta_distribution
+
+    No::
+
+        .. [1] https://en.wikipedia.org/wiki/Zeta_distribution    
+
+
+Use of ``np`` in the "Examples" section
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Do not include ``import numpy as np`` in the code given in the "Examples"
+section.  The NumPy Docstring standard says "The examples may assume that
+``import numpy as np`` is executed before the example code in numpy."  That
+statement makes the import *optional*; this guideline says explicitly
+that the import statement must not be included.
+
+
```

Comment:
```suggestion
How to use Random Number Generator (RNG) and seed
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In case random numbers are needed, `np.random.Generator` API must be used.
Use the helper function to validate the parameter `seed`, `random_state` [TBD: which is preferred?]::

    from scipy._lib._util import check_random_state

    def foo(seed=None):
        rng = check_random_state(seed)
        sample = rng.uniform(...)  # for instance

If you want to write an example using random numbers in the documentation::

    >>> rng = np.random.default_rng()
    >>> sample = rng.uniform(...)  # for instance

.. warning:: Do not specify the seed. This line will get overwritten and the same seed
   will be used for conveniency for all examples.

In tests, you have to use the `np.random.RandomState` interface and here you can use a seed
if you need a deterministic output::

    def test_foo():
        rng = np.random.RandomState(SOME_NUMBER)
        sample = rng.uniform(...)  # for instance

For further info, see `NumPy's documentation
<https://numpy.org/doc/stable/reference/random/index.html>`__.


```

Reply:
@tupui, do you think you could split this into two parts, one for generating random data in the "Examples" section of a docstring, and one for using random data in unit tests?  I hadn't followed very closely the discussions in https://github.com/scipy/scipy/pull/13863, but in https://github.com/scipy/scipy/pull/14182, @rkern pointed me to the important conclusions, including the part about a predefined seed being used when Sphinx builds the docs. This means a developer can test the code that will be used in an example, and be sure that the seed will match the one used in Sphinx.  We need to get some version of [this comment](https://github.com/scipy/scipy/pull/14182#issuecomment-855318533) into these docs.

Reply:
Back end of June. I will get to this as soon as I'm back if this is still a thing.

Reply:
This has been updated and ready to include on my side.

Reply:
I opened #15852 for that.